pkg/hub/submarinerbroker/controller.go

```diff
@@ -115,9 +131,57 @@ func (c *submarinerBrokerController) sync(ctx context.Context, syncCtx factory.S
 	return c.createIPSecPSKSecret(config.SubmarinerNamespace)
 }
 
+func buildBrokerConfig(clusterSet *clusterv1beta1.ManagedClusterSet) (*brokerConfig, error) {
+	config := &brokerConfig{
+		SubmarinerNamespace:         brokerinfo.GenerateBrokerName(clusterSet.Name),
+		Components:                  []string{"service-discovery", "connectivity"},
+		DefaultCustomDomains:        nil,
+		GlobalnetCIDRRange:          "242.0.0.0/8",
+		DefaultGlobalnetClusterSize: 65536,
+		GlobalnetEnabled:            false,
+	}
+
+	if v, ok := clusterSet.Annotations[brokerComponentsAnnotationKey]; ok {
```

Comment:
@skitt @tpantelis do you have any feedback on this approach of using annotations on the ManagedClusterSet object to enable certain features?

Reply:
I don't know if this is a recommended approach and if other apps do it. For the submariner install on the managed clusters, we have a `SubmarinerConfig` CRD to communicate the user config so perhaps we should create an equivalent `BrokerConfig` CRD to be consistent.

Reply:
I'm hesitant about this place, too. 
If `BrokerConfig` is a more acceptable solution, I can try it. @tpantelis @sridhargaddam

Reply:
yeah let's try that. We don't own the `ManagedClusterSet` so our annotations could cause extraneous updates for others and possibly get overwritten.

Reply:
Ideal way to create `Broker` on hub cluster is to install `submariner-operator` on the hub cluster and let the operator handle `broker` creation. `submariner` is created in the spoke in the similar fashion.

Reply:
yeah it seems like we should be able to use the [Broker](https://github.com/submariner-io/submariner-operator/blob/devel/api/submariner/v1alpha1/submariner_types.go#L191) CRD and install the `submariner-operator`.